Source/EasyNetQ.IntegrationTests/Advanced/When_connecting_to_non_existed_vhost.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EasyNetQ.Topology;
+using Xunit;
+
+namespace EasyNetQ.IntegrationTests.Advanced
+{
+    [Collection("RabbitMQ")]
+    public class When_connecting_to_non_existed_vhost : IDisposable
+    {
+        public When_connecting_to_non_existed_vhost(RabbitMQFixture rmqFixture)
+        {
+            Console.WriteLine(rmqFixture.Host);
+            bus = RabbitHutch.CreateBus($"host={rmqFixture.Host};virtualHost=BlaBlaBla;prefetchCount=1;timeout=-1;publisherConfirms=True");
+        }
+
+        public void Dispose() => bus.Dispose();
+
+        private readonly IBus bus;
+
+        [Fact]
+        public async Task Test()
+        {
+            using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(40));
+
+            IExchange e;
+            try
+            {
+                e = await bus.Advanced.ExchangeDeclareAsync("this_exchange_will_never_be_created", c => { }, cts.Token);
```

Comment:
maybe use Assert.Throws instead?

Reply:
I will cancel this PR soon. In the process of debugging, I discovered that the `timeout` parameter was 0 and everything seems to work as intended with an eternal loop. I have other questions about when the eternal loop should break - `PersistentChannel.NeedRethrow`. I asked @Pliner. If we find out that additional fixes are required, then I will make another PR.